### PR TITLE
[ISSUE #3241] Do not use hardcoded gas price

### LINKED ISSUE #3241: Adjust Default Gas price to the lowest Standard instead of using hardcoded 21 Gwei
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a  user, I want to send ETH so that I do not pay high Gas price and I do not need to adjust it manually each time

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: Currently, Gas price is hardcoded to 21 Gwei. We need to adjust it and use lowest Standard price of mainnet.  There is a JSON-RPC method to get the current gas price https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gasprice that can be used

#### Expected behavior
[comment]: # (Describe what you expected to happen.)
Gas price is adjusted to match current lowest Standard price
#### Actual behavior
[comment]: # (Describe what actually happened.)
Gas price is hardcoded and is always 21 Gwei, so user need to go to wallet and manually adjust it for each transaction. 
### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

- Open Status
- Open Wallet
- Send Transaction
- Open Advanced
- Tap on field with fee
-On Transaction Fee screen check Gas Price

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: 0.9.13 and develop build 
* Operating System: Android and iOS

### PR DESCRIPTION
**Pull request by jeluard**
relates to #3241

### Summary:

Rely on geth to access current gas price

### Steps to test:
- Open Status
- Navigate wallet / send
- Verify it's not using default hardcoded value

status: ready

### CODE REVIEW

**Comment by dshulyak** [comment]
Hey, I just want to mention that when we are signing and sending tx in status-go we are always setting suggested (by eth network) gas price and estimated gas limit if they are not provided.

**Comment by jeluard** [comment]
Thanks @dshulyak ! We want to own those values UI side as we have to display it and allow user to modify this default.

File: src/status_im/ui/screens/wallet/events.cljs [hunk 1]
```diff
@@ -202,3 +202,14 @@
                          :content             (i18n/label :t/transactions-delete-content)
                          :confirm-button-text (i18n/label :t/confirm)
                          :on-accept           #(re-frame/dispatch [:wallet/discard-unsigned-transaction transaction-id])}}))
+
+(handlers/register-handler-db
+  :wallet/update-gas-price
+  (fn [{:keys [web3] :as db}]
+    (ethereum/gas-price web3 #(re-frame/dispatch [:wallet/update-gas-price-success %2]))
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/events.cljs]: Shouldn't this code be moved to effects?

**Review by goranjovic — APPROVED** [review]

**Comment by jeluard** [comment]
@rasom Updated to a `fx`

**Comment by janherich** [comment]
@jeluard please resolve conflicts

**Comment by jeluard** [comment]
@janherich done

**Review by janherich — APPROVED** [review]

**Comment by annadanchenko** [comment]
If use Send Transaction from Wallet: I assume that Gas price adjustment should work for mainnet too. I’ve switched to mainnet with ropsten RPC but see 20 Gwei instead of SafeLow 1 Gwei (at least according to https://ethgasstation.info/index.php ) In the same time 1 Gwei is shown ok for testnets.



<blockquote><div><strong><a href="http://ethgasstation.info">ETH Gas Station</a></strong></div></blockquote>

**Comment by jeluard** [comment]
Updated description to make sure related issue is not closed (and bounty not claimed).

**Pull request merged**